server/controllers/gift.controller.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const Gift = require('../../config/db').Gift;
+const messages = require('../helper/messages');
+
+module.exports = {
+  create(req, res) {
+    let assignGift = Object.assign({}, req.body, {event_id: req.params.id});
+
+    Gift.create(assignGift)
+    .then(gift => res.status(201).send(gift))
+    .catch(error => res.status(400).send(error));
+  },
+
+  list(req, res) {
+    console.log(req.params);
+    Gift.findAll({
+      attributes: [
+        'name', 'description', 'link', 'is_available'
+      ],
+      where: {event_id: req.params.id}
+    })
+    .then(gift => res.status(200).send(gift))
+    .catch(error => res.status(400).send(error));
+  },
+
+  retrieve(req, res) {
+    Gift.findById(
+      req.params.giftid, {
+      attributes: [
+          'name', 'description', 'link', 'is_available', 'event_id'
+      ]})
+      .then(gift => {
+        if (gift.dataValues.event_id !== Number(req.params.id) || !gift)  {
+          return res.status(400).send(messages.giftNotFound);
+        } else {
+          return res.status(200).send(gift);
+        }
+      })
+      .catch(error => {
+        return res.status(400).send(error);
+      });
+  },
+
+  update(req, res) {
+    Gift.findById(req.params.giftid).then(gift => {
+      if (gift.dataValues.event_id !== Number(req.params.id) || !gift) {
```

Comment:
create a common function for this validation ( we use here, destroy  and retrieve)